compiler/rustc_hir_analysis/src/collect.rs

```diff
@@ -960,6 +960,7 @@ fn trait_def(tcx: TyCtxt<'_>, def_id: DefId) -> ty::TraitDef {
             }
             Some(items) => items
                 .into_iter()
+                .filter(|item| item.ident().is_some_and(|ident| ident.name != kw::Empty))
```

Comment:
Where's a test that exercises this? There are no tests for `rustc_must_implement_one_of` + AFIT/RPITIT.

Reply:
Can you remove this if we don't need this logic? Or else tell me why we need it? :)

Reply:
Is needed for the `` from trait part

Reply:
How can a user type a `` in the `rustc_must_implement_one_of` list? Isn't this a syntax error?

Reply:
@cjgillot this is coming from RPITITs synthesized as associated types with no names.

Reply:
I should add a comment to mention that

Reply:
`items` is user input, isn't it? How can a user know of synthetic assoc types?

Reply:
Maybe we can make RPITITs not show up in `rustc_must_implement_one_of` somehow?

Reply:
> `items` is user input, isn't it? How can a user know of synthetic assoc types?

We have `fn foo(&self) -> impl Sized;`, so we end lowering to a projection and an associated type that is opaque.
The error says that foo is not implemented but doesn't make any sense to mention that `impl Sized` is not implemented.
We have this thing in item, which is likely to be the projection, so we should avoid those. Maybe I could just filter the ones that are projection to an opaque as @compiler-errors has done in other PRs.

Reply:
You're right, there was something odd going on. Fixed.